### ISSUE TITLE
allow for inPath inside "and" block

### ISSUE THREAD

**Issue by eximius313** [lukas-krecan/JsonUnit#149]
Currently `inPath` is allowed only directly after `assertThatJson`, but having it in `and` block should also be supported.

```
assertThatJson(myJson)
.and(
          a -> a.inPath(path1).isEqualTo("foo"),
          a -> a.inPath(path2).isEqualTo("bar")
);
```
Thank you in advance

**Comment by lukas-krecan** [comment]
Hi, I wanted to do it, but it's surprisingly hard. It would require me to override method `public JsonAssert and(JsonAssertion... assertions)` with something like  `public JsonAssert and(ConfigurableJsonAssertion... assertions)` but I can't figure out how to do it in a usable way (generics and varargs do not mix well). You can try to do it, if you wish. In the meantime you can always do 

```
        ConfigurableJsonAssert json = assertThatJson("{\"key1\":\"foo\",\"key2\":\"bar\"}");
        json.inPath("key1").isEqualTo("foo");
        json.inPath("key2").isEqualTo("bar");
```

**Comment by eximius313** [comment]
@lukas-krecan, I'm still constrained with not being able to use `inPath` within `and`.
Have you maybe figured it out "how to do it in a usable way"?

**Comment by lukas-krecan** [comment]
The best I can do is this https://github.com/lukas-krecan/JsonUnit/pull/204. Would it be enough for you use case?

**Comment by eximius313** [comment]
I'll try and be back with feedback.
When do you plan to release it?

**Comment by lukas-krecan** [comment]
As soon as you will say it's ok. You can try this by copying the method to your project, it's standalone

**Comment by lukas-krecan** [comment]
Released in 2.10.0

**Comment by eximius313** [comment]
It's not good as `and`, because it requires to change the API that Callbacks must be passed to custom assertions, but as a workaround it's fine
Thanks @lukas-krecan
